### PR TITLE
Update UserGuide.md tag

### LINKED ISSUE #159: Include note about tags in UG
Add a note that `t/tagwithoutspace` is a valid tag but `t/tag with space` is an invalid tag

### PR DESCRIPTION
**Pull request by oolongwater**
Closes #159

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -66,6 +67,8 @@ HiredFiredPro is a **desktop app for managing interviews, optimized for use via
 * If you are using a PDF version of this document, be careful when copying and pasting commands that span multiple lines as space characters surrounding line-breaks may be omitted when copied over to the application.
 </box>
 
+* `t/tagwithoutspace` is a valid tag but `t/tag with space` is an invalid tag.
```
Comment (SQ77) [inline docs/UserGuide.md:70]: Can move this to another section "Notes about the parameters"? 
Then can add notes about all the parameters such as NAME, JOB, etc in a table

File: docs/UserGuide.md [hunk 2]
```diff
@@ -81,10 +84,9 @@ Adds a person to HiredFiredPro.
 
 Format: `add n/NAME j/JOB p/PHONE_NUMBER e/EMAIL [s/SKILL]… i/INTERVIEW_SCORE  [t/TAG]…​`
 
-<box type="tip" seamless>
+<box type="tip" seamless></box>
```
Comment (SQ77) [inline docs/UserGuide.md:87]: What is the purpose of this change? Does the box element still work after this?

**Review by SQ77 — COMMENTED** [review]
can also change the `MESSAGE_CONSTRAINTS` in `Tag.java` to `Tags should be alphanumeric and should not contain spaces`

File: docs/UserGuide.md (hunk 1 above)
Comment (oolongwater) [reply docs/UserGuide.md:70]: What notes do you want to add?
